index.d.ts

```diff
@@ -3,12 +3,42 @@ declare interface Logger {
 }
 
 declare type Memory = {
+    /**
+     * Register memory map
+     * @param memoryMap
+     * @example
+     * memory.register({
+     *     key1: 'value1',
+     *     key2: 42
+     *     key3: {
+     *         property: 12
+     *     }
+     * })
+     */
     register(memoryMap: Object): void;
+    /**
+     * Set value to memory
+     * @param key
+     * @param value
+     * @example
+     * memory.setValue('key', 'value');
+     */
     setValue(key: string, value: any): void;
-    getValue(key: string): any;
+    /**
+     * Get value from memory
+     * @param key
+     * @example
+     * const value = memory.getValue('key');
+     */
+    getValue(key: any): any;
```

Comment:
The `key` argument should be string type, shouldtn't is?

Reply:
no, we can pass anything here, e.g in case if the value is processed by cucumber parameter type before.
```typescript
defineParameterType({
    name: 'foo',
    transformer: value => parseInt(value)
})

When('type {foo}', function(x) {
    const value = memory.getValue(x); // x is already number
})
```